lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskTransferrer.java

```diff
@@ -275,7 +301,8 @@ private void updateTransferableTasks(
                 TaskSummaryImpl newSummary = (TaskSummaryImpl) oldSummary.copy();
                 newSummary.setId(oldSummary.getId());
                 newSummary.setExternalId(oldSummary.getExternalId());
-                applyTransferValuesForTask(newSummary, destinationWorkbasket, setTransferFlag);
+                applyTransferValuesForTask(
```

Comment:
Why is applyTransferValuesForTask called twice here?

Reply:
In the first applyTransferValuesForTask, an empty task (no ID,external ID,etc) is set with the transfer values (destinationWorkbasket, owner, etc) and used as a referenceTask. It is then passed to the databank update query so that the tasksToBeUpdated are updated according to the values set in the referenceTask.

But this referenceTask cannot be used as comparison between the old and new task summary in the history event, since the referenceTask does not have the id,externalId,etc of the old task. Which is why the old task has to be copied and the applyTransferValuesForTask has to be called again to set new transferValues to the copy of the task.